### DISCUSSION TITLE
Broken images

### DISCUSSION THREAD

**Discussion by crisacostagit** [community/community#21960] [Other Feature Feedback, Questions, & Ideas]
<p>Hi Guys,</p>
<p>It is first time here and I need immediate help.</p>
<p>I am a newbie coder and I have just recently completed my portfolio. When I first deployed my portfolio on github pages everything worked well and images for my project section were showing but during my last commit the images are now showing as broken. I am using VS code and when I launch the live server the images are showing fine. </p>
<p>here is a snippet:</p>
<p>image on the right is viewing the projects page with live server and the left is thru gitbhub pages.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/6087b52dda042e025679a712d7aafd7db79fbdeb.jpeg" data-download-href="/uploads/short-url/dLWAyyFP81ghwPsmPxWNHB1U3Rp.jpg?dl=1" title="image not working.jpg" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/35595513/181067884-51cf3ea9-85f2-415f-aee7-4053a675b061.jpg" alt="image not working.jpg" data-base62-sha1="dLWAyyFP81ghwPsmPxWNHB1U3Rp" width="690" height="387" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/6087b52dda042e025679a712d7aafd7db79fbdeb_2_690x387.jpg, https://aws1.discourse-cdn.com/github/optimized/1X/6087b52dda042e025679a712d7aafd7db79fbdeb_2_1035x580.jpg 1.5x, https://aws1.discourse-cdn.com/github/original/1X/6087b52dda042e025679a712d7aafd7db79fbdeb.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/6087b52dda042e025679a712d7aafd7db79fbdeb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image not working.jpg</span><span class="informations">1366×768 280 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>If you must check my code to check the problem. here is the repo: <a href="https://github.com/eatcodesforlife/portfolio" rel="nofollow noopener">https://github.com/eatcodesforlife/portfolio </a></p>

**Comment by SethClydesdale** [comment, marked as answer]
<p>Hi <a class="mention" href="/eatcodesforlife">@eatcodesforlife</a>,</p>
<p>It looks like the issue is the two dots plus the slash "…/"at the beginning of your images’ src attributes. This is basically a reference to a folder up one directory from /portfolio/ – it refers to the root of your github pages (<a href="https://eatcodesforlife.github.io/" rel="nofollow noopener">https://eatcodesforlife.github.io/</a>) in this case. Since projects.html is in the same directory as /img, you can refer to it like this :</p>
<pre><code>src="img/flexbox.jpg"
</code></pre>
<p><a href="https://i.imgur.com/8M2KoFD.png" rel="nofollow noopener"><img src="https://i.imgur.com/8M2KoFD.png" alt=""></a></p>
<p> I’m not sure what the result would be on your live server, but you can open the devtools and hover the src to see where it’s getting the images from, like in the example above, if there’s any issues on your end.</p>

**Comment by crisacostagit** [comment]
<p>Hey <a class="mention" href="/sethclydesdale">@sethclydesdale</a>,</p>
<p>That worked! Thanks a mil!</p>

**Comment by iweyouclasses** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c2c56d1f619d88215ba819999a019a6f4ca5cdb5.png" data-download-href="/uploads/short-url/rN1C0FN64rvOp4xDgRdqnYMJkPP.png?dl=1" title="Capture.PNG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/48938876/181067887-6927292e-c732-46b6-9e1d-88a0c463e690.png" alt="Capture.PNG" data-base62-sha1="rN1C0FN64rvOp4xDgRdqnYMJkPP" width="616" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c2c56d1f619d88215ba819999a019a6f4ca5cdb5_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture.PNG</span><span class="informations">633×513 6.11 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <img src="https://user-images.githubusercontent.com/48938876/181067888-8bf36eeb-5504-4787-9036-e03e85383c3c.png" alt="Capture1.PNG" data-base62-sha1="aeGvfyOFZhNdbrYZ7xSW9ZE5jnn" width="616" height="390">
<p>i had tried adding …/  but its look nothing have changed</p>
<p>link :-  <a href="https://iweyouclasses.github.io/core/topics/String.html" rel="nofollow noopener">https://iweyouclasses.github.io/core/topics/String.html</a></p>

**Comment by rustyonrampage** [comment]
<p>I had a similar issue myself, where the images would work on localhost but not on Github pages. <br>
I tried to prefix URLs with … but it broke the links on localhost so I didn’t pushed it to the gh pages. </p>
<p>Interestingly I noticed that the extension of the image was in captial letters PNG whereas path was provided in small letters png. I didn’t had any issues on the localhost but messed up on gh pages which is weirdish. </p>
<p>So I altered the image extension from PNG to png and it worked for that problem.</p>

**Comment by adriansagala** [comment]
<p>Same as mine, in the localhost is JPG but I type jpg (no capital) is Ok, but in github it’s must be in capital. Thanks for the suggestion.</p>

**Comment by mmhashim139** [comment]
<p>Thanks , This solved my issue with images in GitHub pages, I was using …/img while html file in the same directory.</p>

**Comment by KatarzynaMaz** [comment]
<p>I was struggling with the same problem this morning. Images were displaying on local host, but not it GitHub page, so I’m really grateful for this discussion. In my case images in the images folder had lower case ext .jpg, but for some reason I had the capital JPG in the index.html. Worked perfectly on the local host, but GitHub pages didn’t load the images. All fixed now. Thanks again to everyone in this discussion.</p>

**Comment by daniellacornide** [comment]
<p>Hi Everyone!</p>
<p>I am having ussues trying to get my image to show on my live page. The image shows up completely fine on y local file but is broken when I publish.</p>
<p>Ive chedked the code so many times and can’t figure it out. A fresh pair of eyes would be great!</p>
<p>THanks!</p>
<p>Link to code: <a href="https://github.com/daniellacornide/daniellacornide.github.io/blob/c3ad5642b298108fc10403c1790bc2b51a745502/final.html" rel="noopener nofollow ugc">daniellacornide.github.io/final.html at main · daniellacornide/daniellacornide.github.io · GitHub</a></p>

**Comment by airtower-luna** [comment]
<p>Capitalization is different. The <code>src</code> attribute has <code>images/</code>, but the directory in your repository is called <code>Images/</code> with capital <code>I</code>.</p><aside class="onebox githubblob" data-onebox-src="https://github.com/daniellacornide/daniellacornide.github.io/blob/c3ad5642b298108fc10403c1790bc2b51a745502/final.html#L26">
  <header class="source">

      <a href="https://github.com/daniellacornide/daniellacornide.github.io/blob/c3ad5642b298108fc10403c1790bc2b51a745502/final.html#L26" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/daniellacornide/daniellacornide.github.io/blob/c3ad5642b298108fc10403c1790bc2b51a745502/final.html#L26" target="_blank" rel="noopener">daniellacornide/daniellacornide.github.io/blob/c3ad5642b298108fc10403c1790bc2b51a745502/final.html#L26</a></h4>



    <pre class="onebox">      <code class="lang-html">
        <ol class="start lines" start="16" style="counter-reset: li-counter 15 ;">
            <li>  &lt;ul&gt;</li>
            <li>    &lt;li&gt;&lt;a href="final.html"&gt;Home&lt;/a&gt;&lt;/li&gt;</li>
            <li>    &lt;li&gt;&lt;a href="finalwork.html"&gt;Work&lt;/a&gt;&lt;/li&gt;</li>
            <li>    &lt;li&gt;&lt;a href="resume.html"&gt;About Me&lt;/a&gt;&lt;/li&gt;</li>
            <li>  &lt;/ul&gt;</li>
            <li>&lt;/nav&gt;</li>
            <li>
            </li>
<li>&lt;main&gt;</li>
            <li>  &lt;section&gt;</li>
            <li>    &lt;figure&gt;</li>
            <li class="selected">      &lt;img src="images/HS.JPG" alt="Picture of Daniella Cornide."/&gt;</li>
            <li>    &lt;/figure&gt;</li>
            <li>
            </li>
<li>    &lt;article&gt;</li>
            <li>      &lt;h2&gt;Art Director &amp; Designer&lt;/h2&gt;</li>
            <li>    &lt;/article&gt;</li>
            <li>  &lt;/section&gt;</li>
            <li>&lt;/main&gt;</li>
            <li>
            </li>
<li>&lt;footer&gt;</li>
            <li>  &lt;h3&gt;Contact Me&lt;/h3&gt;</li>
        </ol>
      </code>
    </pre>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by daniellacornide** [comment]
<p>I changed it but that didnt seem to work. Not sure what else it could be</p>

**Comment by airtower-luna** [comment]
<p>What doesn’t work? The picture seems to load, though a wall doesn’t fit the alt text. :sweat_smile:</p>

**Comment by jfullstackdev** [comment]
<aside class="quote no-group" data-username="daniellacornide" data-post="9" data-topic="10485">
<div class="title">
<div class="quote-controls"></div>
 daniellacornide:</div>
<blockquote>
<p>Link to code: <a href="https://github.com/daniellacornide/daniellacornide.github.io/blob/c3ad5642b298108fc10403c1790bc2b51a745502/final.html">daniellacornide.github.io/final.html at main · daniellacornide/daniellacornide.github.io · GitHub </a></p>
</blockquote>
</aside>
<p>this is from the link you provided,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/a/1abb1bec5188fca459d6af5eba7db636e832cd36.png" data-download-href="/uploads/short-url/3Oticve17f2ZkwWKGztefWBg8Z0.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181067895-6023a998-4e5a-4500-99f3-907a10ae8645.png" alt="image" data-base62-sha1="3Oticve17f2ZkwWKGztefWBg8Z0" width="690" height="224" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/a/1abb1bec5188fca459d6af5eba7db636e832cd36_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">817×266 15.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<pre><code class="lang-auto">&lt;img src="images/HS.JPG" alt="Picture of Daniella Cornide."/&gt;
</code></pre>
<p>this is from your updated <code>final.html</code> at <code>main</code></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/c/1c776632b84e0ba39904b823370bb1f7367d154f.png" data-download-href="/uploads/short-url/43Pbd0tJglMQFykzH0mb66brsIv.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181067896-8a9fc17e-0c6e-46de-b4a0-01e7ad22bea7.png" alt="image" data-base62-sha1="43Pbd0tJglMQFykzH0mb66brsIv" width="690" height="243" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/c/1c776632b84e0ba39904b823370bb1f7367d154f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">767×271 14.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<pre><code class="lang-auto">&lt;img src="Images/HS.JPG" alt="Picture of Daniella Cornide."/&gt;
</code></pre>
<p>meaning it’s not reflected on <code>main</code> and yes, it will not appear on the live website</p>

**Comment by Elena-Baixy** [comment]
I am struggling with this problem. I think I use the right path. Could someone take a look for me?
It doesn't work neither locally nor on github
`<img src="profile.png" alt="hi"/>`
<img width="431" alt="Screenshot 2023-01-31 at 10 39 41 PM" src="https://user-images.githubusercontent.com/72651795/215939529-bf33c31f-f5cf-428d-b9a9-ecd47b2b99f3.png">
